web/src/search.js

```diff
@@ -11,6 +11,10 @@ import * as search_suggestion from "./search_suggestion";
 export let is_using_input_method = false;
 
 export function narrow_or_search_for_term(search_string, {on_narrow_search}) {
+    if (search_string === "") {
+        message_view_header.exit_search();
+        return "";
+    }
```

Comment:
```
export function exit_search() {                                                             
    const filter = narrow_state.filter();                                                   
    if (!filter || filter.is_common_narrow()) {                                             
        // for common narrows, we change the UI (and don't redirect)                        
        close_search_bar_and_open_narrow_description();                                     
    } else {                                                                                
        // for "searching narrows", we redirect                                             
        window.location.href = filter.generate_redirect_url();                              
    }                                                                                       
    $(".app").trigger("focus");                                                             
}                                                                                           
```

Hmm, is that else clause here going to be produce surprising behavior here?

Reply:
Ah yes it does! I had forgotten that "exit" search is meant to go along with actually performing the search. I'm just going to tack this onto the search redesign PR then, since the code I now want to use is stuff I wrote there.

Reply:
(Well, I guess I'll just wait for it to get merged and then come back to this)